pkg/nirmata/nirmata_ProviderManaged_cluster.go

```diff
@@ -97,15 +97,45 @@ func resourceClusterCreate(d *schema.ResourceData, meta interface{}) error {
 }
 
 func resourceClusterRead(d *schema.ResourceData, meta interface{}) error {
+	apiClient := meta.(client.Client)
+
+	name := d.Get("name").(string)
+
+	clusterID, err := apiClient.QueryByName(client.ServiceClusters, "clustertypes", name)
+	if err != nil {
+		fmt.Println(err.Error())
+		return err
+	}
+	data, err := apiClient.Get(clusterID, &client.GetOptions{})
+	if err != nil {
+		fmt.Printf("failed to retrieve cluster details %s (%s): %v", name, clusterID, err)
+		return err
```

Comment:
should this return an error if we get back a 404 from the server? Or, should it set the ID to "" to indicate it was deleted?